### DISCUSSION TITLE
Zbus - Wait on multiple events in a same thread

### DISCUSSION THREAD

**Discussion by atruffaut** [zephyrproject-rtos/zephyr#64502] [Q&A]
Hello, 
I want to wait for multiple message subscribers in a thread using Zbus module.

I want to have a producer thread publishing to multiple channels. Each channel has a different message subscriber. A consumer thread handling the reading of multiple message subscriber. 

We thought to use Polling API from zephyr to wait on multiple FIFO. Is it possible to use this mechanism to wait on the internal FIFO of a message subscriber ?  

Is there a simpler solution to this problem ? Any suggestion is welcome.

**Comment by nashif** [comment]
@rodrigopex FYI

**Comment by rodrigopex** [comment]
@atruffaut, I suggest you use a single message subscriber to observe all the channels the producer will publish to instead of one message subscriber per channel. After all, the consumer thread will process all the messages sequentially as they are coming. I cannot understand the need to have separated message subscribers if only one thread will process the messages.
